privacyidea/lib/resolvers/HTTPResolver.py

```diff
@@ -0,0 +1,247 @@
+        descriptor['config'] = {
+            'endpoint': 'string',
+            'method': 'string',
+            'headers': 'string',
+            'requestMapping': 'string',
+            'responseMapping': 'string',
+            'hasSpecialErrorHandler': 'bool',
+            'errorResponse': 'string',
+        }
+        return {typ: descriptor}
+
+    @staticmethod
+    def getResolverDescriptor():
+        """
+        return the descriptor of the resolver, which is
+        - the class name and
+        - the config description
+
+        :return: resolver description dict
+        :rtype:  dict
+        """
+        return HTTPResolver.getResolverClassDescriptor()
+
+    def getUserId(self, loginName):
+        """
+        This method only echo's the loginName parameter
+        """
+        return loginName
+
+    def getUsername(self, userid):
+        """
+        This method only echo's the userid parameter
+        """
+        return userid
+
+    def getUserInfo(self, userid):
+        """
+        This function returns all user information for a given user object
+        identified by UserID.
+        :param userid: ID of the user in the resolver
+        :type userid: int or string
+        :return:  dictionary, if no object is found, the dictionary is empty
+        :rtype: dict
+        """
+        return self._getUser(userid)
+
+    def getUserList(self, searchDict=None):
+        """
+        Since it is an HTTP resolver,
+        users are not stored in the database
+        """
+        return []
+
+    def getResolverId(self):
+        """
+        get resolver specific information
+        :return: the resolver identifier string - empty string if not exist
+        """
+        return self.config['endpoint'] if 'endpoint' in self.config else ''
+
+    def loadConfig(self, config):
+        """
+        Load the configuration from the dict into the Resolver object.
+        If attributes are missing, need to set default values.
+        If required attributes are missing, this should raise an
+        Exception.
+
+        :param config: The configuration values of the resolver
+        :type config: dict
+        """
+
+        # Validations
+        getParam(config, 'method', optional=False, allow_empty=False, allowed_values=('get', 'post', 'GET', 'POST'))
+        getParam(config, 'endpoint', optional=False, allow_empty=False)
+        getParam(config, 'requestMapping', optional=False, allow_empty=False)
+        getParam(config, 'responseMapping', optional=False, allow_empty=False)
+        hasSpecialErrorHandler = getParam(
+            config,
+            'hasSpecialErrorHandler',
+            optional=False,
+            default=False,
+            allow_empty=False
+        )
+
+        # Validate special error handler
+        if hasSpecialErrorHandler and not config.get('errorResponse'):
+            getParam(config, 'errorResponse', optional=False, allow_empty=False)
```

Comment:
I should have read the code more closely... the checking is all unnecessary since we receive the data from the database and assume that this data is correct.
So a simple `self.config = config` should suffice here.
Sorry for the trouble.

Reply:
Ok... So validations are not necessary for this resolver or Should I move them to another part?

Reply:
The validations occur when the resolver is created so when we load the configuration from the db we can assume that everything is correct. Simply copying the configuration should be enough here.